flow/Platform.cpp

```diff
@@ -1820,16 +1820,29 @@ bool createDirectory( std::string const& directory ) {
 		if ( mkdir( directory.substr(0, sep).c_str(), 0755 ) != 0 ) {
 			if (errno == EEXIST)
 				continue;
+			auto mkdirErrno = errno;
+
+			// check if directory already exists
+			// necessary due to old kernel bugs
+			struct stat s;
+			const char* dirname = directory.c_str();
+			if (stat(dirname, &s) != -1 && S_ISDIR(s.st_mode)) {
+				TraceEvent("DirectoryAlreadyExists").detail("Directory", dirname).detail("IgnoredError", mkdirErrno);
```

Comment:
Since it is a reasonable pattern to use `createDirectory()` to ensure a directory exists before creating a file in it, something that creates many files could cause this trace event to be spammy, but I think default suppression will take care of it.

Reply:
Default suppression is intended to be a backstop. If something is actually going to be spammy, there should be an explicit suppression for it. Not sure if that's necessary here, though.